### PR TITLE
Add an upgrade test

### PR DESCRIPTION
**Pull request by skitt**
This adds an upgrade test, deploying a cluster with the latest
released version before upgrading it to the current development
version and running the end-to-end tests.

### CODE REVIEW

**Review by mangelajo — APPROVED** [review]
Is this draft anymore? I see it passing

File: Makefile [hunk 1]
```diff
@@ -24,6 +24,12 @@ nettest: package/.image.nettest
 
 e2e: vendor/modules.txt clusters
 
+upgrade-e2e: upgrade e2e
+
+upgrade:
```
Comment (mangelajo) [inline Makefile:29]: May be instead of "upgrade" this could be "deploy-latest" ?

File: .github/workflows/upgrade-e2e.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+---
+name: Upgrade Tests
```
Comment (dfarrell07) [inline .github/workflows/upgrade-e2e.yml:2]:
```suggestion
name: Upgrade
```

File: .github/workflows/upgrade-e2e.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+on:
+  pull_request:
+
+jobs:
+  upgrade-e2e:
+    name: Verify that a cluster upgraded from release continues working
```
Comment (dfarrell07) [inline .github/workflows/upgrade-e2e.yml:9]:
```suggestion
    name: Latest Release to Latest Version
```

File: .github/workflows/upgrade-e2e.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    timeout-minutes: 30
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 0
+      - run: git fetch origin +refs/tags/*:refs/tags/*
```
Comment (dfarrell07) [inline .github/workflows/upgrade-e2e.yml:18]: We need this extra fetching logic?

**Review by dfarrell07 — CHANGES_REQUESTED** [review]

File: .github/workflows/upgrade-e2e.yml (hunk 4 above)
Comment (skitt) [reply .github/workflows/upgrade-e2e.yml]: No, we don’t, I meant to remove it and forgot!

**Review by dfarrell07 — APPROVED** [review]

**Review by tpantelis — APPROVED** [review]

**Comment by mangelajo** [comment]
Late to the party, but :+1:

**Pull request merged**